pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -2947,6 +3208,17 @@ func addOrUpdateClusterDeploymentCondition(cd hivev1.ClusterDeployment,
 	return newConditions
 }
 
+// sanitizeConditions scrubs the condition list for each CD in `cds` so they can reasonably be compared
```

Comment:
Is it necessary for sorting and times be present for comparison anywhere in the file? I have always preferred the asserts/validating functions to not depend on them

Reply:
That's exactly why this is here. It's nulling out the timestamps so we can compare the CD objects using [cmp.Diff](https://github.com/openshift/hive/blob/8671ed2067e47e5ed0b6e718e7163a6741ed7b39/pkg/test/assert/assertions.go#L88). If we were just comparing the conditions, we could explicitly pick out the status/reason/message fields; but extending that to the whole CD object would be impractical.